inference-engine/src/auto_plugin/auto_plugin.cpp

```diff
@@ -61,7 +61,63 @@ IE::IExecutableNetworkInternal::Ptr AutoInferencePlugin::LoadExeNetworkImpl(cons
     }
 
     auto networkPrecision = GetNetworkPrecision(network);
-    return LoadNetworkImpl(network, config, networkPrecision);
+    return LoadNetworkImpl({}, network, config, networkPrecision);
+}
+
+std::shared_ptr<AutoExecutableNetwork> AutoInferencePlugin::LoadNetworkImpl(const std::string& modelPath,
+                                                                            const InferenceEngine::CNNNetwork& network,
+                                                                            const ConfigType& config,
+                                                                            const std::string& networkPrecision) {
+    if (GetCore() == nullptr) {
+        IE_THROW() << "Please, work with AUTO device via InferencEngine::Core object";
+    }
+
+    if (modelPath.empty() && network.getFunction() == nullptr) {
+        IE_THROW() << "AUTO device supports just ngraph network representation";
+    }
+
+    auto fullConfig = mergeConfigs(_config, config);
+    CheckConfig(fullConfig);
+    auto metaDevices = GetDeviceList(fullConfig);
+    auto LoadNetworkAsync =
+        [=](const std::string& device)
```

Comment:
this actually does not solve segment fault, please ignore this change.

Reply:
I was referrring to the lambda capturing being too broad (i.e [=])  not function args/
capturing everything by value (incl 'this') is dangerous, again you have to capture the right things only and only as shared_ptr, e.g. the GetCore should be probably cllaed outside lambda, wrapped to the shared_ptr, and the ptr can be captured by value then, this is exactly you are asking from @ilya-lavrenov to advice in another comment

Reply:
GetCore() is already a raw pointer, that is `InferenceEngine::ICore* core = GetCore();`, I am not sure if it is a good idea to wrap this to shared_ptr. But I agree that we should capture explicit, I will change that. Thanks~

Reply:
and the question I am asking ilya is that if it is OK that ~ICore() happened before any plugin. In AUTO case, ~AUTOPLUGIN() definitely will still call ICore() object. do we have any way to make the lifetime of ICore() as the same with plugin?